nuxeo-features/rest-api/nuxeo-rest-api-test/src/test/java/org/nuxeo/ecm/restapi/test/DocumentSecuredPropertyTest.java

```diff
@@ -0,0 +1,163 @@
+import org.nuxeo.ecm.core.api.PathRef;
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.ACL;
+import org.nuxeo.ecm.core.api.security.ACP;
+import org.nuxeo.ecm.core.api.security.impl.ACPImpl;
+import org.nuxeo.ecm.core.schema.PropertyCharacteristicHandler;
+import org.nuxeo.ecm.core.test.TransactionalFeature;
+import org.nuxeo.ecm.core.test.annotations.Granularity;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.jaxrs.test.CloseableClientResponse;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.Jetty;
+
+/**
+ * Tests the secured property handled by {@link PropertyCharacteristicHandler}.
+ *
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(RestServerFeature.class)
+@Jetty(port = 18090)
+@Deploy("org.nuxeo.ecm.core.api.tests:OSGI-INF/test-documentmodel-secured-types-contrib.xml")
+@RepositoryConfig(cleanup = Granularity.METHOD, init = RestServerInit.class)
+public class DocumentSecuredPropertyTest extends BaseTest {
+
+    public static final String USER_1 = "user1";
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @Inject
+    protected CoreSession session;
+
+    @Before
+    @Override
+    public void doBefore() throws Exception {
+        super.doBefore();
+        ACE ace = ACE.builder(USER_1, READ_WRITE).creator(session.getPrincipal().getName()).isGranted(true).build();
+        ACP acp = new ACPImpl();
+        acp.addACE(ACL.LOCAL_ACL, ace);
+        session.setACP(new PathRef("/folder_2"), acp, false);
+        session.save();
+        txFeature.nextTransaction();
+    }
+
+    @Test
+    public void testAdministratorCanCreate() {
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "path/folder_2",
+                instantiateDocumentBody())) {
+            StatusType status = response.getStatusInfo();
+            assertEquals("HTTP Reason: " + status.getReasonPhrase(), SC_CREATED, status.getStatusCode());
+        }
+    }
+
+    @Test
+    public void testUserCanNotCreate() throws IOException {
+        this.service = getServiceFor(USER_1, USER_1);
+        try (CloseableClientResponse response = getResponse(RequestType.POST, "path/folder_2",
+                instantiateDocumentBody())) {
+            StatusType status = response.getStatusInfo();
+            assertEquals("HTTP Reason: " + status.getReasonPhrase(), SC_BAD_REQUEST, status.getStatusCode());
+            // not necessary to close stream, it will be done by response close
+            JsonNode root = mapper.readTree(response.getEntityInputStream());
+            assertTrue(root.has("message"));
+            JsonNode message = root.get("message");
+            assertTrue(message.isTextual());
+            assertEquals("Cannot set the value of property: scalar since it is readonly", message.getTextValue());
```

Comment:
i think the error check can be done by using org.nuxeo.ecm.restapi.test.BaseTest#getErrorMessage. it's not really required, especially that the code is already backported on 10.10 :)

Reply:
Since code is already backported to 10.10, I won't change it for 8.10 and 9.10. Master (#3138) will receive the update.